app/lib/components/term.js

```diff
@@ -137,19 +172,19 @@ export default class Term extends Component {
     }
 
     if (this.props.foregroundColor !== nextProps.foregroundColor) {
-      this.term.prefs_.set('foreground-color', nextProps.foregroundColor);
+      this.setCssVariable('--foreground-color', nextProps.foregroundColor);
     }
 
     if (this.props.backgroundColor !== nextProps.backgroundColor) {
-      this.term.prefs_.set('background-color', nextProps.backgroundColor);
+      this.setCssVariable('--background-color', nextProps.backgroundColor);
     }
 
     if (this.props.fontFamily !== nextProps.fontFamily) {
       this.term.prefs_.set('font-family', nextProps.fontFamily);
     }
 
     if (this.props.cursorColor !== nextProps.cursorColor) {
-      this.term.prefs_.set('cursor-color', nextProps.cursorColor);
+      this.setCssVariable('--cursor-color', nextProps.cursorColor);
```

Comment:
might have a race condition if nextProps has both new `cursorColor` (or one of the other two colors) and `customCss`, I think the old color might stay put, shouldn't be too hard to fix though